File: 2845_count_of_interesting_subarrays.cpp
class Solution {
public:
    long long countInterestingSubarrays(vector<int>& a, int b, int c) {
        unordered_map<int, long long> d; d[0]++;
        long long e = 0, f = 0;
        for (auto g : a) {
            f += (g % b == c);
            e += d[(f - c + b) % b];
            d[f % b]++;
        }
        return e;
    }
};

File: 2845_count_of_interesting_subarrays.java
class Solution {
    public long countInterestingSubarrays(List<Integer> a, int b, int c) {
        Map<Integer, Long> d = new HashMap<>();
        d.put(0, 1L);
        long e = 0, f = 0;
        for (int g : a) {
            f += (g % b == c) ? 1 : 0;
            e += d.getOrDefault((int)((f - c + b) % b), 0L);
            d.put((int)(f % b), d.getOrDefault((int)(f % b), 0L) + 1);
        }
        return e;
    }
}
